### PR TITLE
FolderMan::checkPathValidityForNewFolder: make sure to work with deleted folders

### PR DESCRIPTION
**Pull request by ogoffart**
Note that we also needed to adjust the server url to contains the user name
in the folder wizard. (As checkPathValidityForNewFolder expect the user name)

Issue #6654

### CODE REVIEW

**Comment by ogoffart** [comment]
If you are wondering why I removed the wording with symlinks, it is because i now always use the canonical path. And there is no need for a different wording with symbolic links.

File: src/gui/folderman.cpp [hunk 1]
```diff
@@ -1306,83 +1306,81 @@ QString FolderMan::trayTooltipStatusString(
...
         if (folderDir.isEmpty()) {
             continue;
         }
-        if (!folderDir.endsWith(QLatin1Char('/'), cs))
-            folderDir.append(QLatin1Char('/'));
-
         const QString folderDirClean = QDir::cleanPath(folderDir) + '/';
-        const QString userDirClean = QDir::cleanPath(path) + '/';
 
         // folderDir follows sym links, path not.
```
Comment (ckamm) [inline src/gui/folderman.cpp]: Since both are canonicalized, that's probably an outdated comments

File: src/gui/folderman.cpp [hunk 2]
```diff
@@ -1306,83 +1306,81 @@ QString FolderMan::trayTooltipStatusString(
...
     Qt::CaseSensitivity cs = Qt::CaseSensitive;
     if (Utility::fsCasePreserving()) {
         cs = Qt::CaseInsensitive;
     }
 
+    const QString userDirClean = QDir::cleanPath(canonicalPath(path)) + '/';
     for (auto i = _folderMap.constBegin(); i != _folderMap.constEnd(); ++i) {
         Folder *f = static_cast<Folder *>(i.value());
-        QString folderDir = QDir(f->path()).canonicalPath();
+        QString folderDir = canonicalPath(f->path());
```
Comment (ckamm) [inline src/gui/folderman.cpp]: Would it be possible to drop this `folderDir` and then drop the `Clean` postfix from userDir and folderDir? I don't see where the non-cleaned variants are used in the first place.

**Comment by ogoffart** [comment]
Thanks for the comments.  (I've updated the commit)

**Comment by ckamm** [comment]
Looks good!

Minor on rereading, feel free to skip:
- maybe move the `+ '/'` inside the `cleanPath()` to make clear there'll not be a double trailing /
- `differentPathes` -> `differentPaths`

**Comment by ogoffart** [comment]
> maybe move the + '/' inside the cleanPath() to make clear there'll not be a double trailing /

cleanPath removes trailing '/'

**Pull request merged**